lighthouse-core/lib/i18n.js

```diff
@@ -43,57 +41,149 @@ const formats = {
 
 /**
  * @param {string} msg
- * @param {Record<string, *>} values
+ * @param {Record<string, *>} [values]
  */
 function preprocessMessageValues(msg, values) {
+  if (!values) return;
+
+  const clonedValues = JSON.parse(JSON.stringify(values));
   const parsed = MessageParser.parse(msg);
   // Round all milliseconds to 10s place
   parsed.elements
     .filter(el => el.format && el.format.style === 'milliseconds')
-    .forEach(el => (values[el.id] = Math.round(values[el.id] / 10) * 10));
+    .forEach(el => (clonedValues[el.id] = Math.round(clonedValues[el.id] / 10) * 10));
 
   // Replace all the bytes with KB
   parsed.elements
     .filter(el => el.format && el.format.style === 'bytes')
-    .forEach(el => (values[el.id] = values[el.id] / 1024));
+    .forEach(el => (clonedValues[el.id] = clonedValues[el.id] / 1024));
+
+  return clonedValues;
+}
+
+/**
+ * @typedef StringUsage
+ * @prop {string} key
+ * @prop {string} template
+ * @prop {*} [values]
+ */
+
+/** @type {Map<string, StringUsage[]>} */
+const formattedStringUsages = new Map();
```

Comment:
this is the new singleton, we might want to put some disclaimers into README somewhere about YMMV using LH as a long-lived module not in a worker. CLI/DevTools/Extension all don't really care about these types of concerns so inevitably we'll make some trade-offs not great for node module users (example: #5683)